src/main/java/com/yahoo/bullet/common/BulletConfig.java

```diff
@@ -398,6 +398,7 @@ public static Validator getValidator() {
     public void merge(Config other) {
         super.merge(other);
         validate();
+        provider = BulletRecordProvider.from(getAs(RECORD_PROVIDER_CLASS_NAME, String.class));
```

Comment:
Good catch but it won't fix it if you do config.set(RECORD_PROVIDER_CLASS_NAME, somethingElse) and do getBulletRecordProvider().

I think the best way to fix it would be to make the getBulletRecordProvider do the check `(provider == null || !provider.getClass.getName().equals(get(RECORD_PROVIDER_CLASS_NAME))` and init if so?